internal/ingress/controller/controller_test.go

```diff
@@ -287,7 +286,7 @@ func TestCheckIngress(t *testing.T) {
 				},
 			}
 
-			if nginx.CheckIngress(ingPath) != nil {
+			if nginx.CheckIngress(ingPath) == nil {
```

Comment:
uh I guess this is wrong per test description:

`When ingress contains invalid path and pathType validation is disabled`

You should probably keep adding `PathAdditionalAllowedChars: "^%$[](){}*+?|",` as it is used to define the allowed chars

Reply:
added back path additional chars and put the test back to != nil